### ISSUE TITLE
ImportError: cannot import name gpuarray

### ISSUE THREAD

**Issue by jagiella** [Theano/libgpuarray#393]
I installed Theano and libgpuarray as described [here](http://deeplearning.net/software/libgpuarray/installation.html). The files are definitely installed:
```
$ ls /usr/local/lib/python2.7/dist-packages/pygpu-0.6.2-py2.7-linux-x86_64.egg/pygpu/
_array.py   blas.c             collectives.c    dtypes.py     _elemwise.pyc   gpuarray.h    __init__.pyc    reduction.pyc  version.pyc
_array.pyc  blas.py            collectives.h    dtypes.pyc    elemwise.pyc    gpuarray.py   numpy_compat.h  tests
basic.py    blas.pyc           collectives.py   _elemwise.c   _elemwise.so    gpuarray.pyc  operations.py   tools.py
basic.pyc   blas.so            collectives.pyc  _elemwise.py  gpuarray_api.h  gpuarray.so   operations.pyc  tools.pyc
blas_api.h  collectives_api.h  collectives.so   elemwise.py   gpuarray.c      __init__.py   reduction.py    version.py
```

When importing theano (as keras backend), I get the following error:
```python
ERROR (theano.gpuarray): pygpu was configured but could not be imported or is too old (version 0.6 or higher required)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/theano/gpuarray/__init__.py", line 21, in <module>
    import pygpu
  File "pygpu/__init__.py", line 7, in <module>
    from . import gpuarray, elemwise, reduction
  File "pygpu/elemwise.py", line 3, in <module>
    from .dtypes import dtype_to_ctype, get_common_dtype
  File "pygpu/dtypes.py", line 6, in <module>
    from . import gpuarray
ImportError: cannot import name gpuarray
```

**Comment by nouiz** [comment]
Hi,

there is a problem in your installation. The simplest if you use conda is
to remove what you installed and install it via conda:

conda install Theano pygpu

This will also install the last release of Thano.

**Comment by jagiella** [comment]
I am not using `conda` but `pip` for the installation of modules. Is there a diverging behavior between the two?

**Comment by nouiz** [comment]
You can only install Theano with pip. It is the same as the Theano on conda
now.

But conda allow us to also install libgpuarray and pygpu, but pip don't
allow us to do the same. So if you want to don't compile gpuarray yourself,
using conda is very useful.

We have limited time to support many configuration setup. We support well
conda, but if you try other setup and it don't work, we don't always have
time to help you make it work.

**Comment by jagiella** [comment]
Well I compiled and installed libgpuarray and pygpu manually, as indicated on the official [website](http://deeplearning.net/software/libgpuarray/installation.html#step-by-step-install-system-library-as-admin)

Nevertheless, it's not working.

**Comment by abergeron** [comment]
Did you reboot since you installed libgpuarray?  If you installed in in /usr/local you might need to run `sudo ldconfig` or reboot before the linker will find it.

If you didn't install it there, you might need to adjust LD_LIBRARY_PATH so that the linker can find the library.

If none of the above applies, please post the output of `ldd /usr/local/lib/python2.7/dist-packages/pygpu-0.6.2-py2.7-linux-x86_64.egg/pygpu/gpuarray.so`.

**Comment by mo-morikawa** [comment]
I have the same issue as @jagiella after installing both libgpuarray adn pygpu from the official site, and here is the out of for the command _ldd /usr/local/lib/python2.7/dist-packages/pygpu-0.6.2-py2.7-linux-x86_64.egg/pygpu/gpuarray.so_

```
linux-vdso.so.1 =>  (0x00007ffeb39fe000)
	libgpuarray.so.2 => /usr/local/lib/libgpuarray.so.2 (0x00007ff140451000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007ff140234000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ff13fe6a000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007ff13fc66000)
	/lib64/ld-linux-x86-64.so.2 (0x00005580fc0ff000)

```

**Comment by mo-morikawa** [comment]
I have the same issue as @jagiella after installing both libgpuarray adn pygpu from the official site, and here is the out of for the command _ldd /usr/local/lib/python2.7/dist-packages/pygpu-0.6.2-py2.7-linux-x86_64.egg/pygpu/gpuarray.so_

```
linux-vdso.so.1 =>  (0x00007ffeb39fe000)
	libgpuarray.so.2 => /usr/local/lib/libgpuarray.so.2 (0x00007ff140451000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007ff140234000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ff13fe6a000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007ff13fc66000)
	/lib64/ld-linux-x86-64.so.2 (0x00005580fc0ff000)

```

**Comment by nouiz** [comment]
great. So I'll close this issue.

**Comment by abergeron** [comment]
Just for future googlers:  The reboot probably ran `ldconfig` as part of the booting scripts which added the new library to the cache.  This is documented in the linux-specific notes in the installation documentation.

**Comment by kirkhansen** [comment]
> I managed to fix the problem by re-installing theano and then rebooting the system.

I just ran into this and some other similar errors. It appears the stable version of Theano in pip will barf all kinds of errors making you think your cuda / nvidia or even the pygpu install is broken. My fix was to `pip install -U git+https://github.com/Theano/Theano.git#egg=Theano`

**Comment by loretoparisi** [comment]
@kirkhansen yours was the only working solution!!!

**Comment by mattprice09** [comment]
After spending more time than I'd like to admit with this issue, @kirkhansen had the solution!

**Comment by nouiz** [comment]
We just released 1.0 on it is now on pip. So this should be better now.
